server/src/main/java/org/elasticsearch/common/time/DateFormatters.java

```diff
@@ -345,7 +345,7 @@
     );
 
     private static final DateTimeFormatter BASIC_YEAR_MONTH_DAY_FORMATTER = new DateTimeFormatterBuilder()
-        .appendValue(ChronoField.YEAR, 4, 4, SignStyle.NORMAL)
+        .appendValue(ChronoField.YEAR, 4, 10, SignStyle.NORMAL)
```

Comment:
I think we should have a separate BASIC_YEAR_MONTH_DAY_PRINTER
FORMATTER is a bit unfortunate name to be fair, as it is used for parsing..

Reply:
👍